### PR TITLE
Feature/69 basic client UI

### PR DESCRIPTION
**Pull request by maxdiebold-erg**
## Related Issues:
* [EQ-69](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=865&view=detail&selectedIssue=EQ-69)
* [EQ-71](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=865&view=detail&selectedIssue=EQ-71)

## Main Changes:
* Added state management for the query inputs that will be available on the interface.
* Created React components using the USWDS library.
* Added the capability of updating the page URL when inputs change. Input fields are populated by parameters in the initial URL.
* Verified the EPA template being used is current.
   * A custom `max-width` value was set on the header, which doesn't match the official template, but if that is desired then it can be added back.

## Steps To Test:
1. Navigate to [http://192.168.0.130:3000/attains#dataProfile=catchmentCorrespondence&fileFormat=json&state=AR&state=GA](http://192.168.0.130:3000/attains#dataProfile=catchmentCorrespondence&fileFormat=json&state=AR&state=GA) and confirm that the fields are filled properly. In the displayed query boxes at the bottom of the page, the `fileFormat` parameter should be included in the _Current Query_ box, but not in the _Catchment Correspondence API Query_ box.
2. Clear a state from the _State_ input and confirm the URL updates.
3. Use a gibberish string for the file format, such as [http://192.168.0.130:3000/attains#dataProfile=assessmentUnitsMonitoring&fileFormat=cursive](http://192.168.0.130:3000/attains#dataProfile=assessmentUnitsMonitoring&fileFormat=cursive), and confirm it is replaced with the default format CSV.
4. Check that other design choices adhere to the US web design standards.

### CODE REVIEW

File: app/client/src/routes/home.tsx [hunk 1]
```diff
@@ -1,16 +1,511 @@
+import { useCallback, useEffect, useReducer, useRef, useState } from 'react';
+import Select from 'react-select';
 // components
-import { Loading } from "components/loading";
+import Alert from 'components/alert';
+import Button from 'components/button';
```
Comment (cschwinderg) [inline app/client/src/routes/home.tsx:5]:
```suggestion
```

Part of a comment above. I don't think this component is necessary at this time.

File: app/client/src/routes/home.tsx [hunk 2]
```diff
@@ -1,16 +1,511 @@
...
+                label="Don't show again on this computer"
+                onChange={(_ev) => setDontShowAgain(!dontShowAgain)}
+                styles={['margin-right-1 margin-y-auto']}
+              />
+              <Button
+                onClick={() => setIntroVisible(false)}
+                styles={['margin-top-2']}
+              >
+                Close Intro
+              </Button>
```
Comment (cschwinderg) [inline app/client/src/routes/home.tsx:386]:
```suggestion
              <Button
                onClick={() => setIntroVisible(false)}
                className="usa-button margin-top-2"
              >
                Close Intro
              </Button>
```

Part of comment above. I don't think this component is necessary at this time.

File: app/client/src/routes/home.tsx [hunk 3]
```diff
@@ -1,16 +1,511 @@
...
+                >
+                  Download
+                </Button>
+                <Button
+                  color="white"
+                  onClick={(_ev) => inputDispatch({ type: 'reset' })}
+                  styles={['margin-x-auto']}
+                >
+                  Clear Search
+                </Button>
```
Comment (cschwinderg) [inline app/client/src/routes/home.tsx:436]:
```suggestion
                <button
                  onClick={() => {}}
                  className="usa-button margin-x-auto margin-bottom-1"
                >
                  Download
                </button>
                <button
                  onClick={(_ev) => inputDispatch({ type: 'reset' })}
                  className="usa-button usa-button--outline margin-x-auto"
                >
                  Clear Search
                </button>
```

Part of comment above. I don't think this component is necessary at this time.

File: app/client/src/routes/home.tsx [hunk 4]
```diff
@@ -1,16 +1,511 @@
...
+const controlFields = ['dataProfile', 'fileFormat'];
+
+/*
+## Utilities
+*/
+// Converts a JSON object into a parameter string
+function buildQueryString(query: URLQueryState, includeControl = true) {
+  const paramsList: Array<URLQueryParam> = [];
+  Object.entries(query).forEach(([field, value]) => {
+    if (!includeControl && controlFields.includes(field)) return;
```
Comment (cschwinderg) [inline app/client/src/routes/home.tsx:63]:
```suggestion
    if (!includeControl && field === 'dataProfile') return;
```

I think we want to include the `fileFormat` parameter in the API URL. 

There may be a better way to accomplish this, but the above suggestion will work.

File: app/client/src/routes/home.tsx [hunk 5]
```diff
@@ -1,16 +1,511 @@
...
+type URLQueryArg = string | number | boolean;
+
+type InputValue = URLQueryArg | URLQueryArg[] | null;
+
+/*
+## Constants
+*/
+const defaultFileFormat = 'csv';
+
+const controlFields = ['dataProfile', 'fileFormat'];
```
Comment (cschwinderg) [inline app/client/src/routes/home.tsx:54]:
```suggestion
const controlFields = ['dataProfile', 'format'];
```

The expert query endpoints will except `f` or `format`. I think we should atleast shorten the `fileFormat` parameter down to just `format`.

**Review by cschwinderg — CHANGES_REQUESTED** [review]
This is looking great so far!

I left some suggestions below. 

I think we need to have a discussion, during today's meeting, about how we want the queries to be specified in the urls. Below are some examples that compare and contrast what I have seen in some of the stuff we use. 

* `?assessmentUnitId='abc1234'&assessmentUnitId='1234Abc'&assessmentUnitId='xyz987'` - We currently use this in the EQ UI and the water quality portal does this as well.
* `?assessmentUnitId=[%22abc1234%22,%221234Abc%22,%22xyz987%22]` - This was something I did on the server side to have a more concise URL. This was quick and easy to get the queries working as all I needed to do was `JSON.parse(req.query.<paramName>)` to get the array out.
* `?assessmentUnitId=abc1234,1234Abc,xyz987` - This is what the attains web services use. Very concise and probably what we should be using to keep consistent with attains. This might be a little tricky though if any of our input values have commas. 
* `?where=assessmentUnitId%20in%20(%27abc1234%27%2C%271234Abc%27%2C%27xyz987%27)` - This is what esri uses. We probably don't want to use this for our purposes.

File: app/client/src/components/accordion.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import type { ReactNode } from 'react';
```
[the remark this answers is not in the document]
Comment (maxdiebold-erg) [reply app/client/src/components/accordion.tsx:1]: Good point, I'll get rid of that.

**Comment by cschwinderg** [comment]
I commited a change to the dev build workflow to include the `--legacy-peer-deps` flag, which is needed for installing `react-select`.

**Review by cschwinderg — APPROVED** [review]
Looks good! Thank you for making those changes.

**Pull request merged**